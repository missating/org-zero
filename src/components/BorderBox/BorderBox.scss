.borderbox-container {
  // border-top: 5px solid;
  // border-left: 1px solid;
  // flex-grow: 0;
  // min-width: 40px;
  // flex-shrink: 0;

  &__heading,
  &__text
  {
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
    margin: 0;
    font-size: 0.6rem;
  }
}
