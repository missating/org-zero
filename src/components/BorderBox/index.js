// react library
import React from 'react'
// third-party libraries
import PropTypes from 'prop-types'
import { createUseStyles } from 'react-jss'

// styles
import './BorderBox.scss'

const useStyles = createUseStyles({
  borderBox: props => ({
    borderTop: {
      width: 5,
      style: 'solid'
    },
    borderLeft: {
      width: 1,
      style: 'solid'
    },
    width: props.boxWidth,
    flexGrow: 0,
    flexShrink: 0,
    paddingTop: props.padding,
    borderColor: props.colors,
    height: props.boxHeight,
    // width: props.boxWidth,

    '@media (max-width: 600px)': {
      paddingTop: 5
    }
  }),
})



const BorderBox = ({ data, name, colors, padding, boxHeight, boxWidth }) => {
  // const divStyle = {
  //   borderColor: colors,
  //   width: boxSize,
  //   height: boxHeight,
  //   paddingTop: padding
  // }
  const classes = useStyles({ colors, padding, boxHeight, boxWidth })

  return (
    <article className={classes.borderBox}>
      <h2 className="borderbox-container__heading">
        {name}
      </h2>
      <p className="borderbox-container__text">
        {data[0].firstName + ' ' + data[0].lastName}
      </p>
      <p className="borderbox-container__text">
        {data.length} {data.length > 1 ? `employees` : `employee`}
      </p>
    </article>
  )
}

export default BorderBox

BorderBox.propTypes = {
  data: PropTypes.arrayOf(PropTypes.object).isRequired,
  name: PropTypes.string.isRequired,
  colors: PropTypes.string.isRequired,
  // boxWidth: PropTypes.string.isRequired,
  padding: PropTypes.string.isRequired,
  boxHeight: PropTypes.string.isRequired
}
