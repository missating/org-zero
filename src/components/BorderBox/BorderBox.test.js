// react library
import React from 'react';

// third-party libraries
import { shallow } from 'enzyme';

// components
import BorderBox from './index'

describe('BorderBox Component', () => {
  const initialProps = {
    data: [{
      employeeId: "1",
      firstName: "Aiden",
      lastName: "Chung",
      jobTitle: "First Lord of intervieworg",
      officeLocation: "Paris",
      department: "intervieworg",
      managerEmployeeId: null,
      reports: [{}]
    }],
    name: 'Test Department',
    colors: '#00000',
    boxSize: '0.032133676092544985%',
    boxHeight: '51px',
    padding: '51px'
  }
  it('should be rendered properly', () => {

    const wrapper = shallow(<BorderBox {...initialProps} />);
    expect(wrapper.find('article').length).toEqual(1);
    expect(wrapper.find('h1').length).toBe(0)
    expect(wrapper).toMatchSnapshot();
  });
})

