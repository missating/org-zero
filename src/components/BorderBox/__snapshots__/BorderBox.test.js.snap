// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BorderBox Component should be rendered properly 1`] = `ShallowWrapper {}`;
